pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/realtime_to_offline_segments/RealtimeToOfflineSegmentsTaskExecutor.java

```diff
@@ -197,7 +197,7 @@ public void preProcess(PinotTaskConfig pinotTaskConfig) {
     Preconditions.checkState(inputSegmentsDir.mkdirs(), "Failed to create input directory: %s for task: %s",
         inputSegmentsDir.getAbsolutePath(), taskType);
     for (File indexDir : originalIndexDirs) {
-      FileUtils.copyDirectoryToDirectory(indexDir, inputSegmentsDir);
+      FileUtils.moveDirectoryToDirectory(indexDir, inputSegmentsDir, false);
```

Comment:
Ideally we should not move the index directories because the caller might still need them. We can directly read from them instead of copying them to another dir.
The current code should work because the parent class won't access the original index directories, but please at least add a TODO to describe the side effect

Reply:
I can give it a shot in this PR, or would you prefer to do it separately?

Reply:
@oker1 `RealtimeToOfflineSegmentsTaskExecutorTest` failed. Please check what causes the failure.
Making it in one PR is even better